tower-batch/tests/ed25519.rs

```diff
@@ -127,5 +141,29 @@ async fn batch_flushes_on_max_latency() -> Result<()> {
     // Use a very high max_items and a short timeout to check that
     // flushing is happening based on hitting max_latency.
     let verifier = Batch::new(Ed25519Verifier::new(), 100, Duration::from_millis(500));
-    timeout(Duration::from_secs(1), sign_and_verify(verifier, 10)).await?
+    timeout(Duration::from_secs(1), sign_and_verify(verifier, 10, None)).await?
+}
+
+#[tokio::test]
+async fn fallback_verification() -> Result<()> {
+    zebra_test::init();
+
+    let verifier = Fallback::new(
+        // Because the Batch wrapper was changed to have a generic error type,
+        // we have to specify it explicitly to avoid an inference hole.
+        // This is pretty unergonomic -- from the error message it's very unintuitive
+        // that the right fix is to change `Batch::new` to `Batch::<_,_,Stuff>::new`.
+        Batch::<_, _, Box<dyn std::error::Error + Send + Sync + 'static>>::new(
```

Comment:
@yaahc It seems like the generic error type changes in #500 cause a type inference failure here.  Is there a way to avoid having to write this type hint?  If not, I think it might be better to roll back to using `Box<dyn Error>` for `Batch`.

Reply:
@hdevalence yea, I just implemented the fix in #666